nifi-provenance-reporting-tasks/src/main/java/io/egm/nifi/reporting/ElasticsearchProvenanceReporter.java

```diff
@@ -85,56 +92,77 @@ public final List<PropertyDescriptor> getSupportedPropertyDescriptors() {
         descriptors = super.getSupportedPropertyDescriptors();
         descriptors.add(ELASTICSEARCH_URL);
         descriptors.add(ELASTICSEARCH_INDEX);
+        descriptors.add(PROCESSORS_TYPE_ALLOWLIST);
         return descriptors;
     }
 
     public void indexEvents(final List<Map<String, Object>> events, final ReportingContext context) throws IOException {
         final String elasticsearchUrl = context.getProperty(ELASTICSEARCH_URL).getValue();
         final String elasticsearchIndex = context.getProperty(ELASTICSEARCH_INDEX).evaluateAttributeExpressions().getValue();
         final ElasticsearchClient client = getElasticsearchClient(elasticsearchUrl);
+        final List<String> processorTypes =
+                Arrays.asList(context.getProperty(PROCESSORS_TYPE_ALLOWLIST).getValue().split(","));
+
         events.forEach(event -> {
             final String id = Long.toString((Long) event.get("event_id"));
 
             if (!event.containsKey("process_group_name") || !event.containsKey("component_name")) {
                 getLogger().warn("Provenance event has no process group or processor, ignoring");
                 return;
             }
-
-            Map<String, Object> preparedEvent = new HashMap<>();
-            preparedEvent.put("event_id", event.get("event_id"));
-            preparedEvent.put("event_time_millis", event.get("event_time"));
-            preparedEvent.put("event_time_iso_utc", event.get("event_time_iso_utc"));
-            preparedEvent.put("event_type", event.get("event_type"));
-            preparedEvent.put("component_type", event.get("component_type"));
-            preparedEvent.put("component_url", event.get("component_url"));
-            preparedEvent.put("component_name", event.get("component_name"));
-            preparedEvent.put("process_group_name", event.get("process_group_name"));
-            preparedEvent.put("process_group_id", event.get("process_group_id"));
-            preparedEvent.put("status", event.get("status"));
-            preparedEvent.put("download_input_content_uri", event.get("download_input_content_uri"));
-            preparedEvent.put("download_output_content_uri", event.get("download_output_content_uri"));
-            preparedEvent.put("view_input_content_uri", event.get("view_input_content_uri"));
-            preparedEvent.put("view_output_content_uri", event.get("view_output_content_uri"));
-            try {
-                preparedEvent.put("updated_attributes", objectMapper.writeValueAsString(event.get("updated_attributes")));
-                preparedEvent.put("previous_attributes", objectMapper.writeValueAsString(event.get("previous_attributes")));
-            } catch (JsonProcessingException e) {
-                getLogger().error("Error while writing value of previous or updated attributes, ignoring them", e);
+            if(!event.containsKey("component_type")) {
```

Comment:
I am wondering if it can happen...

Reply:
According to https://github.com/easy-global-market/nifi-provenance-reporting-bundle/blob/0569290539db32649160c4d2863d580715cdfb38/nifi-provenance-reporting-tasks/src/main/java/io/egm/nifi/reporting/AbstractProvenanceReporter.java#L168 it can happen..?

Reply:
yes, I know, this is why I was only wondering if it can really happen

Reply:
anyway, leave it as it is.